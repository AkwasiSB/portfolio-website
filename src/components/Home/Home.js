import PropTypes from 'prop-types';
import React from 'react';
import { FiChevronDown } from 'react-icons/fi';
import styles from './Home.scss';

export default class Home extends React.Component {
	static propTypes = {
		isHeaderSticky: PropTypes.bool.isRequired,
	}

	render() {
		const buttonMiscStyle = this.props.isHeaderSticky ? 'scale-down': '';

		return (
			<section id='home' className={styles.home}>
				<div className={styles.homeImage}>
					<h3 className={styles.imageText}>Hi Visitor,</h3>

					<h1 className={styles.imageText}>So let's get started</h1>

					<p className={styles.imageText}>I'm a remote Frontend Developer based in Ghana</p>

					<a className={styles.imageButton} href='#about'>WANT MORE Â»</a>
				</div>

				<a
					className={`${styles.imageButton} ${styles.scrollButton} ${buttonMiscStyle}`}
					href='#about'>
					<FiChevronDown size={64} />
				</a>
			</section>
		);
	}
}
