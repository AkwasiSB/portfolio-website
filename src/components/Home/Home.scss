$screen-sm: 768px;
$screen-md: 1200px;

.home {
	@media (min-width: $screen-sm) {
		h1 {
			font-size: 5.4rem;
		}

		h3 {
			font-size: 3.1rem;
		}
	}
}

.home-image {
	background-image: linear-gradient(to right, rgba(128, 0, 128, 0.65)), url('../../images/caspar-camille-rubin-669071-unsplash.jpg');
	background-position: center;
	background-repeat: no-repeat;
	background-size: cover;
	position: relative;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: flex-start;
	height: 100vh;

	@media (min-width: $screen-sm) {
		background-image: linear-gradient(to right, rgba(228, 101, 114, 0.7)), url('../../images/christopher-gower-291246-unsplash.jpg');
		height: 59rem;
		padding-left: 10rem;
	}

	@media (min-width: $screen-md) {
		padding-left: 12.8rem;
	}
}

.image-text {
	color: #fff;
	padding: 0 2.5rem;
	word-wrap: break-word;
	margin: 1.5rem 0;
}

h1.image-text {
	margin: 3rem 0 2rem;
}

.image-button {
	color: #fff;
	background: transparent;
	font-size: 1.4rem;
	font-weight: bold;
	padding: 1rem 1.7rem;
	margin: 3.5rem 2.5rem;
	border: 1px solid #fff;
}

:local(.scroll-button) {
	composes: image-button;
	border: none;
	margin: 0;
	position: absolute;
	bottom: 3rem;
	left: 50%;
	transform: translate(-50%, 0);
	width: 6rem;
	height: 6rem;
	text-align: center;
	animation: Floating 2s infinite ease-in-out;

	:global &.scale-down {
		bottom: 0;
	}

	@media (min-width: $screen-sm) {
		bottom: 4.5rem;
		
		:global &.scale-down {
			bottom: 4.5rem;
		}
	}

	@media (min-width: $screen-md) {
		bottom: 12rem;
		
		:global &.scale-down {
			bottom: 12rem;
		}
	}
}

@keyframes Floating{
	from {
		transform: translate(-50%, 0);
	}
	50% {
		transform: translate(-50%, 2rem);
	}
	to {
		transform: translate(-50%, -0);
	}
}
