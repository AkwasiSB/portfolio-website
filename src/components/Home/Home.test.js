import Home from './Home';

it('should render <Home /> correctly', () => {
    const homeWrapper = shallow(<Home isHeaderSticky={false} />);

    expect(homeWrapper).toMatchSnapshot();
});
