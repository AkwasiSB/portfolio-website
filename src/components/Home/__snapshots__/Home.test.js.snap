// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Home /> correctly 1`] = `
<section
  className="home"
  id="home"
>
  <div
    className="homeImage"
  >
    <h3
      className="imageText"
    >
      Hi Visitor,
    </h3>
    <h1
      className="imageText"
    >
      So let's get started
    </h1>
    <p
      className="imageText"
    >
      I'm a remote Frontend Developer based in Ghana
    </p>
    <a
      className="imageButton"
      href="#about"
    >
      WANT MORE Â»
    </a>
  </div>
  <a
    className="imageButton scrollButton "
    href="#about"
  >
    <FiChevronDown
      size={64}
    />
  </a>
</section>
`;
